### PR TITLE
Add test 'Send a message with an ordered product'

### PR DESCRIPTION
**Pull request by Roomsmaniia**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.7.x
| Description?      | Add test 'Send a message with an ordered product'
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | TEST_PATH="functional/FO/03_userAccount/01_orderHistory/01_orderDetails" npm run specific-test
| Possible impacts? | No impact

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23736)

### CODE REVIEW

File: tests/UI/pages/FO/myAccount/orderDetails.js [hunk 1]
```diff
@@ -23,6 +28,13 @@ class OrderHistory extends FOBasePage {
   isOrderReturnFormVisible(page) {
     return this.elementVisible(page, this.orderReturnForm, 1000);
   }
+
+  async addAMessage(page, messageOption) {
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/orderDetails.js]: Please add a method description

File: tests/UI/pages/FO/myAccount/orderDetails.js [hunk 2]
```diff
@@ -6,9 +6,14 @@ class OrderHistory extends FOBasePage {
     super();
 
     this.pageTitle = 'Order details';
+    this.successMessageText = 'Message successfully sent';
+    this.messageSend = 'Test';
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/myAccount/orderDetails.js]: It should be better added to the test, 
And I think that you can use `campaign/faker/contactUs.js` to create a new message.

```suggestion
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+const helper = require('@utils/helpers');
+const loginCommon = require('@commonTests/loginBO');
+const {Statuses} = require('@data/demo/orderStatuses');
+
+// Import data
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+const {DefaultCustomer} = require('@data/demo/customer');
+
+// Importing pages
+
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]:
```suggestion
// Importing pages
// FO
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+const foHomePage = require('@pages/FO/home');
+const foLoginPage = require('@pages/FO/login');
+const foMyAccountPage = require('@pages/FO/myAccount');
+const foOrderHistoryPage = require('@pages/FO/myAccount/orderHistory');
+const productPage = require('@pages/FO/product');
+const cartPage = require('@pages/FO/cart');
+const checkoutPage = require('@pages/FO/checkout');
+const orderConfirmationPage = require('@pages/FO/checkout/orderConfirmation');
+const orderDetails = require('@pages/FO/myAccount/orderDetails');
+const dashboardPage = require('@pages/BO/dashboard');
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]:
```suggestion
const orderDetails = require('@pages/FO/myAccount/orderDetails');

// BO
const dashboardPage = require('@pages/BO/dashboard');
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+    it('should sign out from FO', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'sighOutFO', baseContext);
+
+      await orderConfirmationPage.logout(page);
+      const isCustomerConnected = await orderConfirmationPage.isCustomerConnected(page);
+      await expect(isCustomerConnected, 'Customer is connected').to.be.false;
+    });
+  });
+  describe('Check invoice file in BO', async () => {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]:
```suggestion
  });
  
  describe('Check invoice file in BO', async () => {
```

Add empty line for readability

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+    it(`should update order status to '${Statuses.paymentAccepted.status}'`, async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'updateOrderStatus', baseContext);
+
+      const textResult = await ordersPage.setOrderStatus(page, 1, Statuses.paymentAccepted);
+      await expect(textResult).to.equal(ordersPage.successfulUpdateMessage);
+
+      const orderStatus = await ordersPage.getTextColumn(page, 'osname', 1);
+      await expect(orderStatus, 'Order status was not updated').to.equal(Statuses.paymentAccepted.status);
+    });
+    it('disconnect from BO', async function () {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]:
```suggestion
    });

    it('disconnect from BO', async function () {
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+      await loginCommon.logoutBO(this, page);
+    });
+  });
+  describe('Send a message on order history', async () => {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]:
```suggestion
  });

  describe('Send a message on order history', async () => {
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+    it('Go to order details ', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToFoToOrderDetails', baseContext);
+
+      await foOrderHistoryPage.goToDetailsPage(page);
+
+      const successMessageText = await orderDetails.addAMessage(page, messageOption);
+      await expect(successMessageText).to.equal(orderDetails.successMessageText);
+    });
+  });
+  describe('Check message in BO', async () => {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]:
```suggestion
  });

  describe('Check message in BO', async () => {
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+      await dashboardPage.goToSubMenu(
+        page,
+        dashboardPage.customerServiceParentLink,
+        dashboardPage.customerServiceLink,
+      );
+
+      const pageTitle = await customerServicePage.getPageTitle(page);
+      await expect(pageTitle).to.contains(customerServicePage.pageTitle);
+    });
+    it('should check customer name', async function () {
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]:
```suggestion
    });

    it('should check customer name', async function () {
```

File: tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+const orderDetails = require('@pages/FO/myAccount/orderDetails');
+const dashboardPage = require('@pages/BO/dashboard');
+const ordersPage = require('@pages/BO/orders/index');
+const customerServicePage = require('@pages/BO/customerService/customerService');
+
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_FO_userAccount_orderHistory_orderDetails_sendMessage';
+const messageOption = 'Hummingbird printed t-shirt - Size : S- Color : White';
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/FO/03_userAccount/01_orderHistory/01_orderDetails/01_sendMessage.js]: I think this shouldn't be hard coded.

**Review by SD1982 — DISMISSED** [review]

File: tests/UI/pages/FO/myAccount/orderDetails.js (hunk 2 above)
Comment (Roomsmaniia) [reply tests/UI/pages/FO/myAccount/orderDetails.js]: Replace by random value

**Review by boubkerbribri — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — CHANGES_REQUESTED** [review]
## WAITING FOR 1773 RELEASE

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**